addons/transifex/models/ir_translation.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from os.path import join as opj
+import os
+import re
+import werkzeug
+
+from odoo import models, fields, tools
+from odoo.modules.module import ad_paths
+
+
+class IrTranslation(models.Model):
+
+    _inherit = 'ir.translation'
+
+    transifex_url = fields.Char("Transifex URL", compute='_get_transifex_url')
+
+    def _get_transifex_url(self):
+        """ Construct transifex URL based on the module on configuration """
+        # e.g. 'https://www.transifex.com/odoo/'
+        base_url = self.env['ir.config_parameter'].sudo().get_param('transifex.project_url')
+
+        tx_config_files = ""
+        for ad_path in ad_paths:
+            tx_path = opj(opj(ad_path, '.tx'), 'config')
+            if os.path.isfile(tx_path):
+                with open(tx_path, 'r') as tx_file:
+                    tx_config_files += tx_file.read()
+
+            # parent directory ad .tx/config is root directory in odoo/odoo
+            tx_path = opj(opj(opj(ad_path, os.pardir), '.tx'), 'config')
+            if os.path.isfile(tx_path):
+                with open(tx_path, 'r') as tx_file:
+                    tx_config_files += tx_file.read()
+
+        if not base_url or not tx_config_files:
+            self.transifex_url = False
+        else:
+            base_url = base_url.rstrip('/')
+
+            # will probably be the same for all terms, avoid multiple searches
+            translation_languages = list(set(self.mapped('lang')))
+            languages = self.env['res.lang'].with_context(active_test=False).search(
+                [('code', 'in', translation_languages)])
+            language_codes = dict((l.code, l.iso_code) for l in languages)
+
+            # .tx/config files contains the project referenche
+            # using ini files like '[odoo-master.website_sale]'
+            translation_modules = list(set(self.mapped('module')))
+            project_modules = {}
+            for module in translation_modules:
+                p = re.compile('\[([\w\-]+)\.%s\]' % module)
```

Comment:
That's completely unnecessary since we're only using the compiled pattern once. Just call `re.findall(pattern, string)`.